### ISSUE TITLE
Be able to import types when using `NodeNext` TypeScript module resolution

### ISSUE THREAD

**Issue by ChocolateLoverRaj** [tj/commander.js#1703]
I'm making an npm package which uses `ora`. To use `ora`, I switched to `NodeNext` module resolution. After doing that, the type checking for `commander` stopped working. I think `package.json` needs to be updated to be compatible with `NodeNext` module resolution. 

Here is the guide to by the author of `ora`: https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c.
Here is some related TypeScript documentation: https://devblogs.microsoft.com/typescript/announcing-typescript-4-5-beta/#esm-nodejs

**Comment by ChocolateLoverRaj** [comment]
I found out how to make it work: I copied `typings/index.d.ts` to `esm.d.mts`

**Comment by shadowspawn** [comment]
How are you importing Commander into your source file? e.g.

```
import { Command } from 'commander';
```

**Comment by ChocolateLoverRaj** [comment]
Creating `esm.d.mts` with this also works:
```js
export * from './typings/index.js'
```

**Comment by ChocolateLoverRaj** [comment]
@shadowspawn I'm importing it exactly like that

**Comment by shadowspawn** [comment]
Looks like we are missing `"types"` entry in `"exports"` in `package.json` for (next) TypeScript.

- https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing
- https://github.com/microsoft/TypeScript/issues/46281

**Comment by ChocolateLoverRaj** [comment]
@shadowspawn Should I make a PR adding the `"types"`?

**Comment by shadowspawn** [comment]
Yes, if you like. I was about to myself, but give you first option. 😄

**Comment by shadowspawn** [comment]
Fixed in Commander v9.2.0
